### ISSUE TITLE
NotFoundError: [_Derived_]No gradient defined for op: Einsum on Tensorflow 1.1.5

### ISSUE THREAD

**Issue by TheGlobalist** [google-research/bert#1012]
I'm using Tensorflow 1.15.2 for making a WSD system, made with BERt in the Embeddings Layer.
This is the code that I use for the model

```
input_word_ids = tf.keras.layers.Input(shape=(64,), dtype=tf.int32, name="input_word_ids")
input_mask = tf.keras.layers.Input(shape=(64,), dtype=tf.int32, name="input_mask")
segment_ids = tf.keras.layers.Input(shape=(64,), dtype=tf.int32, name="segment_ids")
# BERt = BERtLayer()([input_word_ids, input_mask, segment_ids])
bert = hub.KerasLayer("https://tfhub.dev/tensorflow/bert_multi_cased_L-12_H-768_A-12/1", trainable=True)
pooled_output, sequence_output = bert([input_word_ids, input_mask, segment_ids])
# self.vocab_file = bert.resolved_object.vocab_file.asset_path.numpy()
# self.do_lower_case = bert.resolved_object.do_lower_case.numpy()
LSTM = tf.keras.layers.Bidirectional(
    tf.keras.layers.LSTM(
        units=hidden_size,
        dropout=dropout,
        recurrent_dropout=recurrent_dropout,
        return_sequences=True,
        return_state=True
    )
)(sequence_output)
LSTM = self.attention_layer(LSTM)
```

Keras, though, on training, raises the following exception. How can I fix this?

```
NotFoundError: [_Derived_]No gradient defined for op: Einsum
     [[{{node Func/_36}}]]
     [[training/SGD/gradients/gradients/keras_layer/cond/StatefulPartitionedCall_grad/PartitionedCall/gradients/StatefulPartitionedCall_grad/PartitionedCall/gradients/StatefulPartitionedCall_grad/SymbolicGradient]]
```

**Comment by TheGlobalist** [comment]
This issue now also happen in TF 2.X

**Comment by induraj2020** [comment]
having same issue:
NotFoundError:  [_Derived_]No gradient defined for op: StatefulPartitionedCall
	 [[{{node Func/_4}}]]
	 [[PartitionedCall/gradients/StatefulPartitionedCall_grad/PartitionedCall/gradients/StatefulPartitionedCall_grad/SymbolicGradient]] [Op:__inference_distributed_function_76126]

Function call stack:
distributed_function

**Comment by hammadrizwan** [comment]
Any fixes for this:
NotFoundError:  [_Derived_]No gradient defined for op: StatefulPartitionedCall
	 [[{{node Func/_4}}]]
	 [[PartitionedCall/gradients/StatefulPartitionedCall_grad/PartitionedCall/gradients/StatefulPartitionedCall_grad/SymbolicGradient]] [Op:__inference_distributed_function_48070]

Function call stack:
distributed_function

**Comment by TheGlobalist** [comment]
> Any fixes for this:
> NotFoundError: [_Derived_]No gradient defined for op: StatefulPartitionedCall
> [[{{node Func/_4}}]]
> [[PartitionedCall/gradients/StatefulPartitionedCall_grad/PartitionedCall/gradients/StatefulPartitionedCall_grad/SymbolicGradient]] [Op:__inference_distributed_function_48070]
> 
> Function call stack:
> distributed_function

No... If it is the same issue that happened with ALBERT a while ago, it seems that Google needs to rebuild the model

**Comment by hammadrizwan** [comment]
Using the model directly by using hub.KerasLayer works fine but by defining a custom layer and selecting specific layers to fine tune gives error.

**Comment by TheGlobalist** [comment]
> Using the model directly by using hub.KerasLayer works fine but by defining a custom layer and selecting specific layers to fine tune gives error.

In my case, the error is triggered if I load BERT from hub.KerasLayer and set it as trainable

**Comment by hammadrizwan** [comment]
Yes setting trainable=True triggers the error.

**Comment by hammadrizwan** [comment]
> 
> 
> > Using the model directly by using hub.KerasLayer works fine but by defining a custom layer and selecting specific layers to fine tune gives error.
> 
> In my case, the error is triggered if I load BERT from hub.KerasLayer and set it as trainable

https://github.com/tensorflow/hub/issues/540

**Comment by arnoegw** [comment]
https://tfhub.dev/tensorflow/bert_multi_cased_L-12_H-768_A-12/1 is not exported from this GitHub repo, but from https://github.com/tensorflow/models (hence "tensorflow", not "google"). I'm discussing the same issue filed as  https://github.com/tensorflow/hub/issues/540 but I recommend to close this issue here as not in scope.

**Comment by jarumil** [comment]
> having same issue:
> NotFoundError: [_Derived_]No gradient defined for op: StatefulPartitionedCall
> [[{{node Func/_4}}]]
> [[PartitionedCall/gradients/StatefulPartitionedCall_grad/PartitionedCall/gradients/StatefulPartitionedCall_grad/SymbolicGradient]] [Op:__inference_distributed_function_76126]
> 
> Function call stack:
> distributed_function

you have the solution?

**Comment by RobRomijnders** [comment]
[I had a similar issue. Not sure if my solution applies. ]

I realized my hub module has several tags. Trying a more appropriate tag did have the gradient op. 

Here's a code sample that you could quickly try:
For example, my module has tag '"A"' and tag '"B"'. For tag "B", the gradient op is defined, then the following should work:

```python
with tf.GradientTape() as tape:
  some_input = tf.random.uniform(shape=(13, 224, 224, 3))

  model = hub.KerasLayer("/path/to/hub", tags='B')

  some_output = model(some_input)
  loss = tf.reduce_mean((some_output)**2)

grad = tape.gradient(loss, im)
print(grad)
```

Running the same with tag '"A"' will fail with the same error.
